distributed/tests/test_worker.py

```diff
@@ -947,28 +1001,67 @@ def __sizeof__(self):
     assert result.data == 123
 
 
+class Bad:
+    def __init__(self, size):
+        self.size = size
+
+    def __getstate__(self):
+        raise TypeError()
+
+    def __sizeof__(self):
+        return self.size
+
+
 @gen_cluster(client=True)
 async def test_fail_write_to_disk(c, s, a, b):
-    class Bad:
-        def __getstate__(self):
-            raise TypeError()
 
-        def __sizeof__(self):
-            return int(100e9)
-
-    future = c.submit(Bad)
+    # Key immediately stored to disk, fails to serialize
+    future = c.submit(Bad, int(100e9))
     await wait(future)
 
     assert future.status == "error"
 
-    with pytest.raises(TypeError):
+    with pytest.raises(TypeError, match="Could not serialize"):
         await future
 
     futures = c.map(inc, range(10))
     results = await c._gather(futures)
     assert results == list(map(inc, range(10)))
 
 
+@gen_cluster(
+    client=True,
+    nthreads=[("", 1)],
+    worker_kwargs={"memory_limit": 2000},
+    config={
+        "distributed.worker.memory.target": False,
+        "distributed.worker.memory.spill": 0.5,
+    },
+)
```

Comment:
you're hitting the pause threshold. Tasks go through only thanks to a race condition.

Reply:
@crusaderky I modify the test and included your suggestion of pulling the "distributed.spill" logs instead of "distributed.protocol.pickle" logs, but this causes the test to fail since the logs are empty. Are we pulling the right logs? or the problem is somewhere else?